### PR TITLE
Bump Knative/K8s dependencies

### PR DESCRIPTION
**Pull request by mattmoor**
# Changes

This incorporates the breaking changes from K8s 1.18 client libs through cli.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Run the code checkers with `make check`
- [ ] Regenerate the manpages, docs and go formatting with `make generated`
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```release-note
NONE
```

/hold

### CODE REVIEW

**Comment by mattmoor** [comment]
This depends on:
 * pipelines: https://github.com/tektoncd/pipeline/pull/3319
 * triggers: https://github.com/tektoncd/triggers/pull/784

**Comment by mattmoor** [comment]
At least the unit test failures here should be fixed by: https://github.com/tektoncd/pipeline/pull/3328

**Comment by mattmoor** [comment]
I am hoping that https://github.com/tektoncd/pipeline/pull/3348 fixes the manifestation of the new `structured-merge-diff` problems here 🤞

**Comment by mattmoor** [comment]
Alright, it does...

**Review by vdemeester — APPROVED** [review]
/cc @danielhelfand @piyush-garg @pradeepitm12 @chmouel

File: go.mod
Comment (piyush-garg) [inline go.mod:20]: is it fine to bump to a commit in master branch, we may need first to bump to 0.17 branch and then do the release based on 0.17.x

File: go.mod
Comment (piyush-garg) [inline go.mod:45]: WIP /hold

File: go.mod
Comment (vdemeester) [reply go.mod:20]: I think it's fine, we don't necessary need to do a release for each pipeline release. but, to be discussed

**Comment by mattmoor** [comment]
/hold

We need to merge the upstream PRs, this is working off my fork!

File: go.mod
Comment (mattmoor) [reply go.mod:45]: FWIW, we have a Github action that will flag `DO NOT SUBMIT` lines and add a failing check to the PR.

Happy to send y'all a PR with that if you have any interest?  @vdemeester @ImJasonH @bobcatfish @afrittoli

**Comment by mattmoor** [comment]
/hold cancel

Alright, I believe all of the dependent PRs have landed and this is RFAL.

**Review by danielhelfand — APPROVED** [review]
I'll let one more set of eyes take a look, but looks good to me.

**Comment by mattmoor** [comment]
@danielhelfand fwiw, @vdemeester had a look above before my upstream PRs landed (and I dealt with merge conflicts).

**Comment by mattmoor** [comment]
cc @n3wscott reviewed a lot of the Knative 0.18 migration PRs

**Pull request merged**
